src/test/fuzz/addrman.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <addrdb.h>
+#include <addrman.h>
+#include <chainparams.h>
+#include <merkleblock.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <time.h>
+#include <util/asmap.h>
+
+#include <cstdint>
+#include <optional>
+#include <string>
+#include <vector>
+
+void initialize()
+{
+    SelectParams(CBaseChainParams::REGTEST);
+}
+
+void test_one_input(const std::vector<uint8_t>& buffer)
+{
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+
+    SetMockTime(ConsumeTime(fuzzed_data_provider));
+    CAddrMan addr_man;
+    if (fuzzed_data_provider.ConsumeBool()) {
+        while (fuzzed_data_provider.ConsumeBool()) {
+            addr_man.m_asmap = BytesToBits(ConsumeRandomLengthByteVector(fuzzed_data_provider));
+        }
```

Comment:
any reason to re-assign this in a loop?

Reply:
Also to make seeds re-usable it should probably use the same encoding like the other places. `git grep 'asmap = ConsumeRan'`

Reply:
Oh, the loop was a bug. Sorry about that! Thanks for catching.

Now using the newly introduced `ConsumeRandomLengthBitVector(...)` also in `src/test/fuzz/net`.